nixos/tests/nixos-test-driver/lib-extend.nix

```
@@ -0,0 +1,27 @@
+let
+  pkgs = import ../../.. {
+    overlays = [
+      (new: old: {
+        lib = old.lib.extend (self: super: {
+          sorry_dave = "sorry dave";
+        });
+      })
+    ];
+  };
+  inherit (pkgs.testers) nixosTest;
+  sampleEvaluation = nixosTest {
```

```suggestion
  inherit (pkgs.testers) runNixOSTest;
  sampleEvaluation = runNixOSTest {
```

No awkward `callPackage` dependency injection, and this one supports `imports` and any other module syntax. It also pretty much guarantees that the `pkgs` in the nodes is going to be equal to the `pkgs` passed and catches mistaken `nixpkgs.*` definitions.

I guess I'll improve the docs...

---

well found the same bug in `runNixOSTest` then :D

---

I'm not comfortable using only `runNixOSTest` as the bugs currently happens when either are used and I need both to be fixed ideally.
I adjusted the tests accordingly, let me know what you think.